example-app/src/main/java/com/adyen/checkout/example/ui/googlepay/compose/SessionsGooglePayScreen.kt

```diff
@@ -118,7 +117,9 @@ private fun SessionsGooglePayContent(
 
             is SessionsGooglePayState.ShowButton -> {
                 PayButton(
-                    onClick = onButtonClicked,
+                    onClick = {
+                        googlePayComponent.startGooglePayScreen(googlePayLauncher)
+                    },
                     allowedPaymentMethods = googlePayComponent.getGooglePayButtonParameters().allowedPaymentMethods,
```

Comment:
Currently, this line is crashing when there is an action after the payment (for example 3DS2 was triggered), because `googlePayComponent` is null. When an action is triggered I think this method will recompose, which will cause `googlePayComponent` to be null again. 

One way I can think of to mitigate this issue is to use `var googlePayComponent: GooglePayComponent? by remember { mutableStateOf(null) }`. I'm not very experienced with compose though, so I'm not sure if this is a proper solution. How do you think we can tackle this issue?

Reply:
That's right, we need to `remember` this value. Let me propose something.

Reply:
I'm worried about using an optional type (and needing to null check every call to `googlePayComponent`) when we know the lifecycle of the variable well (it gets initialized at the beginning, and can be relied upon from that point on).
I feel that some of that is due to the composable views and the google pay component lifecycles not being in tune (as in, the component does things that don't necessarily concern the view, like initializing itself when the component data is received).
Two initial ideas I see to alleviate the impact from the component in the view are:
- Keeping the component and some of its operation in a viewModel, that surfaces UI-related state changes to the UI when necessary.
- Decoupling some initialization parameters in the component that are not needed right away. For example some of the first actions issued by the google pay component is `startGooglePayScreen`. This method does not required any of the parameters fetched at init time.

Reply:
> Keeping the component and some of its operation in a viewModel, that surfaces UI-related state changes to the UI when necessary.

Not sure if this is really possible. Our components extend `ViewModel`. Having a reference from a `ViewModel` to, in this case, `GooglePayComponent` sounds like bad practice and could potentially cause memory leaks.

> Decoupling some initialization parameters in the component that are not needed right away. For example some of the first actions issued by the google pay component is startGooglePayScreen. This method does not required any of the parameters fetched at init time.

Yes, I think this the way to go. Moving the initializing of the component up in the tree could probably resolve the nullability issue of my previous proposal as well. I was thinking to even move the component to the activity, but maybe that goes against any compose best practices:

<img width="549" alt="image" src="https://github.com/Adyen/adyen-android/assets/17701279/32b14a81-4b0f-4605-9c33-50857c8b88c8">